### PR TITLE
Onboarding to V3 publishing 

### PR DESCRIPTION
**Pull request by epananth**
Fixes #

### Context

Onboarding to V3 publishing. 
Link to what is V3 -> https://github.com/dotnet/arcade/blob/master/Documentation/CorePackages/Publishing.md#what-is-v3-publishing-how-is-it-different-from-v2

### Testing

CI build - https://dev.azure.com/devdiv/DevDiv/_build/results?buildId=4391278&view=results
Promotion pipeline - https://dev.azure.com/devdiv/DevDiv/_build/results?buildId=4391300&view=results

### Notes

Hardcoding of BUILD_REPOSITORY_URI is temporary fix. This will be reverted once this is fixed https://github.com/dotnet/arcade/issues/6827

### CODE REVIEW

**Comment by epananth** [comment]
Asset manifest sample 

image

File: eng/common/templates/job/publish-build-assets.yml [hunk 1]
```diff
@@ -53,10 +53,17 @@ jobs:
         downloadPath: '$(Build.StagingDirectory)/Download'
       condition: ${{ parameters.condition }}
       continueOnError: ${{ parameters.continueOnError }}
-    
+
     - ${{ if and(eq(parameters.runAsPublic, 'false'), ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
       - task: NuGetAuthenticate@0
 
+    - task: PowerShell@2
```
Comment (BenVillalobos) [inline eng/common/templates/job/publish-build-assets.yml:60]: My understanding has typically been that anything under `eng/common` is "stuff arcade owns". Will this be overridden if arcade bumps up a version?

Presumably the fix would be in far before we even do that, but I'm curious.

File: .vsts-dotnet.yml [hunk 2]
```diff
@@ -77,6 +77,13 @@ stages:
       inputs:
         dropName: $(VisualStudio.DropName)
 
+    - task: PowerShell@2
+      inputs: # This is a temporary fix to get msbuild onboarded with v3 publishing. This will be resolved soon ->https://github.com/dotnet/arcade/issues/6827
+        targetType: 'inline'
+        script: |
+          Write-Host "Overwriting BUILD_REPOSITORY_URI."
+          Write-Host "##vso[task.setvariable variable=BUILD_REPOSITORY_URI;]https://dnceng@dev.azure.com/dnceng/internal/_git/dotnet-msbuild"
```
Comment (BenVillalobos) [inline .vsts-dotnet.yml:85]: Was BUILD_REPOSITORY_URI being used before?

**Review by mmitche — APPROVED** [review]

**Review by BenVillalobos — COMMENTED** [review]
Mostly LGTM. So the permanent change here really just setting PublishingVersion to 3 in `Publishing.props`, and all other changes are temp workarounds to get that to work right?

File: eng/common/templates/job/publish-build-assets.yml (hunk 1 above)
Comment (epananth) [reply eng/common/templates/job/publish-build-assets.yml:60]: This fix will block arcade update till I get the other fix out. I am going to be working on it next week. I will have to coordinate with the arcade-services roll out. I will keep you posted on that.

File: .vsts-dotnet.yml (hunk 2 above)
Comment (epananth) [reply .vsts-dotnet.yml:85]: It is used when generating the asset manifest and during publishing build assets...

**Review by BenVillalobos — APPROVED** [review]
LGTM. We're going to have to stop taking arcade updates until the temporary solution gets a permanent fix. We shouldn't have a problem with that, considering we've been on 1.0 until now.

One thing that could potentially break us is the signing issue we've had in the past that required us to take an arcade update.

**Pull request merged**